iceoryx2-cli/iox2-config/src/main.rs

```diff
@@ -43,9 +43,18 @@ fn main() -> Result<()> {
             .install();
     }
 
-    set_log_level(LogLevel::Warn);
+    set_log_level_from_env_or(LogLevel::Warn);
+
+    let cli = match Cli::try_parse() {
+        Ok(cli) => cli,
+        Err(e) => {
+            // --help and --version is treated as a parse error by clap
+            // printing the error is actually printing the result of those commands ...
+            let _ = e.print();
+            return Ok(());
+        }
+    };
```

Comment:
I'm 99% sure that what you do here is equivalent to

```suggestion
    let cli = Cli::parse();
```

The only difference is the return code. With your code, it wil be 0 but with `parse` it will be non 0 when an unknown command is used, e.g. `--foo`.

Can you check if I'm right?

Reply:
It worked.